eclipse-che/che-operator
# feat: native user authentication

Signed-off-by: Michal Vala <mvala@redhat.com>



### What does this PR do?

This is adding support for new authentication without Keycloak. Currently it's only for OpenShift using OpenShift OAuth. It's using oauth-proxy (https://github.com/openshift/oauth-proxy) in the gateway.

Technically, it's adding `auth.nativeUserMode` field in the CR. When enabled, che-operator does not deploy keycloak AND deploys the gateway with authentication and authorization capabilities.


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19705


### How to test this PR?

I keep latest image of this PR here `quay.io/mvala/che-operator:gh19705-newAuth`

Deploy with this patch:
```
apiVersion: org.eclipse.che/v1
kind: CheCluster
metadata:
  name: eclipse-che
spec:
  auth:
    nativeUserMode: true
```

and deploy templates from https://github.com/sparkoo/che-operator/tree/gh19705-newAuth/deploy

with `chectl server:deploy --platform=openshift --installer=operator --workspace-engine=dev-workspace --che-operator-image=quay.io/mvala/che-operator:gh19705-newAuth --che-operator-cr-patch-yaml=/tmp/che_cr.yaml --templates=/tmp/cheop-templates`

Chectl will fail waiting for a keycloak, which is not deployed in nativeUserMode. However, che-operator will deploy che successfully underneath. Get the route `oc get route -n eclipse-che` and open it in browser as usual. You should be redirected to OpenShift login page, so login as any OpenShift user. After login, you should get immediately to the dashboard and you can start any devworkspace.

To logout, you need to delete site cookies, this is not done yet.


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
